src/main/java/io/cryostat/agent/FlightRecorderHelper.java

```diff
@@ -55,17 +56,41 @@ public Recording createRecordingWithCustomTemplate(String template)
 
     public Optional<TemplatedRecording> createRecordingWithPredefinedTemplate(
             String templateNameOrLabel) {
-        Optional<Configuration> opt = getTemplate(templateNameOrLabel);
-        if (opt.isEmpty()) {
-            log.error(
-                    "Cannot start recording with template named or labelled {}",
-                    templateNameOrLabel);
-            return Optional.empty();
+        if ("ALL".equals(templateNameOrLabel)) {
+            Map<String, String> allTemplate = new HashMap<>();
+            FlightRecorder.getFlightRecorder()
+                    .getEventTypes()
+                    .forEach(
+                            e -> {
+                                e.getSettingDescriptors()
+                                        .forEach(
+                                                s -> {
+                                                    allTemplate.put(
+                                                            e.getName() + "#" + s.getName(),
+                                                            s.getDefaultValue());
+                                                });
+                                allTemplate.put(e.getName() + "#enabled", "true");
+                            });
+
+            Recording recording = new Recording(allTemplate);
+            recording.setToDisk(true);
+
+            return Optional.of(new TemplatedRecording(null, recording));
```

Comment:
I don't think this `null` is safe. There is only one place that calls `templatedRecording.getConfiguration()`, and it assumes that it will be non-null:

https://github.com/cryostatio/cryostat-agent/blob/c28859afd1a5d8b210b52dfac3dba370c26880b8/src/main/java/io/cryostat/agent/CryostatClient.java#L376

Since this is the only accessor of the `configuration`, and all it actually wants from it is the name, I think we can do some light refactoring:

```patch
diff --git a/src/main/java/io/cryostat/agent/CryostatClient.java b/src/main/java/io/cryostat/agent/CryostatClient.java
index a6d8719..9a77e9c 100644
--- a/src/main/java/io/cryostat/agent/CryostatClient.java
+++ b/src/main/java/io/cryostat/agent/CryostatClient.java
@@ -37,6 +37,7 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Executor;
 import java.util.function.Function;
 
+import io.cryostat.agent.FlightRecorderHelper.ConfigurationInfo;
 import io.cryostat.agent.FlightRecorderHelper.TemplatedRecording;
 import io.cryostat.agent.WebServer.Credentials;
 import io.cryostat.agent.harvest.Harvester;
@@ -49,7 +50,6 @@ import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import jdk.jfr.Configuration;
 import jdk.jfr.Recording;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.input.CountingInputStream;
@@ -373,8 +373,8 @@ public class CryostatClient {
         Instant start = Instant.now();
         String timestamp = start.truncatedTo(ChronoUnit.SECONDS).toString().replaceAll("[-:]", "");
         String template =
-                opt.map(TemplatedRecording::getConfiguration)
-                        .map(Configuration::getName)
+                opt.map(TemplatedRecording::getConfigurationInfo)
+                        .map(ConfigurationInfo::getName)
                         .map(String::toLowerCase)
                         .map(String::trim)
                         .orElse("unknown");
diff --git a/src/main/java/io/cryostat/agent/FlightRecorderHelper.java b/src/main/java/io/cryostat/agent/FlightRecorderHelper.java
index 65d9217..82a5058 100644
--- a/src/main/java/io/cryostat/agent/FlightRecorderHelper.java
+++ b/src/main/java/io/cryostat/agent/FlightRecorderHelper.java
@@ -89,7 +89,8 @@ public class FlightRecorderHelper {
             Recording recording = new Recording(configuration.getSettings());
             recording.setToDisk(true);
 
-            return Optional.of(new TemplatedRecording(configuration, recording));
+            return Optional.of(
+                    new TemplatedRecording(new ConfigurationInfo(configuration), recording));
         }
     }
 
@@ -125,15 +126,15 @@ public class FlightRecorderHelper {
 
     @SuppressFBWarnings(value = {"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
     public static class TemplatedRecording {
-        private final Configuration configuration;
+        private final ConfigurationInfo configuration;
         private final Recording recording;
 
-        public TemplatedRecording(Configuration configuration, Recording recording) {
-            this.configuration = configuration;
-            this.recording = recording;
+        public TemplatedRecording(ConfigurationInfo configuration, Recording recording) {
+            this.configuration = Objects.requireNonNull(configuration);
+            this.recording = Objects.requireNonNull(recording);
         }
 
-        public Configuration getConfiguration() {
+        public ConfigurationInfo getConfigurationInfo() {
             return configuration;
         }
 
@@ -142,6 +143,40 @@ public class FlightRecorderHelper {
         }
     }
 
+    public static class ConfigurationInfo {
+        private final String name;
+        private final String label;
+        private final TemplateType type;
+
+        public ConfigurationInfo(TemplateType type, String name, String label) {
+            this.type = Objects.requireNonNull(type);
+            this.name = Objects.requireNonNull(name);
+            this.label = Objects.requireNonNull(label);
+        }
+
+        public ConfigurationInfo(Configuration configuration) {
+            this(TemplateType.TARGET, configuration.getName(), configuration.getLabel());
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getLabel() {
+            return label;
+        }
+
+        public TemplateType getType() {
+            return type;
+        }
+    }
+
+    public enum TemplateType {
+        TARGET,
+        CUSTOM,
+        ;
+    }
+
     @SuppressFBWarnings(value = "URF_UNREAD_FIELD")
     public static class RecordingInfo {
 

```